search-provider/eks-metadata-provider.c

```diff
@@ -614,6 +614,21 @@ clear_gparameter_with_allocated_name (GParameter *parameter)
   g_value_unset (&parameter->value);
 }
 
+/* Add "limit" if it isn't present, otherwise xapian-bridge
+ * will complain. */
+static GVariant *
+add_default_query_parameters (GVariant *query_parameters)
+{
+  g_autoptr(GVariantDict) vardict = g_variant_dict_new (query_parameters);
+
+  if (!g_variant_dict_contains (vardict, "limit"))
+    {
+      g_variant_dict_insert (vardict, "limit", "u", G_MAXUINT)
```

Comment:
Hmm, this used to work...
Which query exactly does not have the limit, and which is the commit that broke it?
Why not use `SENSIBLE_LIMIT` instead of `G_MAXUINT`?

Reply:
The issue here is that any caller of the metadata interface can omit the "limit" parameter and it will get its default value (which will be 0 on SDK1 and 2, G_MAXUINT on SDK3 and 4).

I think the relevant caller of the metadata interface in this case is `/content_data` -> it needs to work out the sets that the content belongs to so it can render the metadata section correctly: https://github.com/endlessm/eos-companion-app-integration/blob/master/eoscompanion/ekn_content_adjuster.py#L221